src/tasks/generatePublishTx.ts

```diff
@@ -75,9 +78,9 @@ export function generatePublishTx({
           } catch (e) {
             if (e.message.includes("Could not resolve name")) {
               try {
-                const registryAddress = await provider.resolveName(
-                  ensName.split(".").slice(1).join(".")
-                );
+                const registryAddress = await new ethers.JsonRpcProvider(
+                  ethereumUrl
+                ).resolveName(ensName.split(".").slice(1).join("."));
```

Comment:
Maybe this could be a util function like `getRegistryFromENS`?

Reply:
sounds good, makes sense to have it on the toolkit